python/ql/src/experimental/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -120,6 +128,14 @@ module EssaFlow {
       nodeFrom.(EssaNode).getVar() = p.getAnInput()
```

Comment:
In these three cases, C# takes any of the _last_ reads of the input variable as `nodeFrom`, and only if there are no reads do we take the SSA node. I believe this will currently not work
```
if (..)
  x = taint;
  clean(x);
else
  x = taint;
  clean(x)
sink(x)
```
because you will jump directly from both definitions of `x` to the call to `sink`.

Reply:
Even worse, something like this will (I believe) also not work:
```
x = ...
if (...)
   x.Foo = taint;
else
   x = ...
sink(x.Foo)
```
because there is not step from the `x` in `x.Foo = taint` to the phi node for `x` after the if-then-else.

Reply:
@yoff have you added the testcases from above somewhere, and checked how we handle them after use-use flow? :blush:

Reply:
Actually, I think the latter case will work, as long as the store step in `x.Foo = taint` targets the refined SSA node for `x`. But if it instead targets the post-update node for `x` (as for Java and C#), the change is needed (and it will be needed for the first case anyway).

Reply:
Not yet (but very soon), the need is tracked [here](https://github.com/github/codeql-python-team/issues/133). I think we probably need to remove some essa-flow and let use-use do the work.